### ISSUE TITLE
Add support for configuring optimizer and scheduler in `PCDLit`

### ISSUE THREAD

**Issue by adosar** [adosar/aidsorb#25]
Currently, `Adam` is hard-coded which limits the ability to tune hyperparameters easily from a script.

Suggested changes:
* Add parameter `optimizer` and `lr_scheduler`
* These would be tuplese of the form `(name, config`)

This is a **breaking change**.

> [!NOTE]
> This doesn't affect the CLI, since optimizer and scheduler can still be specified in the `config.yaml` keys `'optimizer'` and `'lr_scheduler'`, respectively. That is, this change adds flexibility when training from a script.

**Comment by adosar** [comment]
Note that in `configure_optimizers`, parameters for which `requires_grad=False` are excluded from optimization. This is for safety: https://discuss.pytorch.org/t/does-optimizer-skip-update-of-parameters-with-requires-grad-false/216205
